### PR TITLE
Added dot product from Annoy c++ source

### PR DESCRIPTION
**Pull request by benjamin-james**
I noticed Annoy has the dot product distance and wrote a quick drop-in interface. I don't know if this breaks things downstream but wanted to see if there was interest in adding this to the API.

### CODE REVIEW

File: local/tests/runit.dotproduct.R [hunk 1]
```diff
@@ -0,0 +1,92 @@
+
```
Comment (eddelbuettel) [inline local/tests/runit.dotproduct.R:1]: We don't need this file.  Check `git blame` on the others -- this was for the predecessor to `tinytest`.

**Review by eddelbuettel — APPROVED** [review]
This looks very good!  

Would you mind adding a ChangeLog entry (date/name/email with two spaces between them, tab to '*' describing file changed and how, one can group as needed).

**Comment by eddelbuettel** [comment]
> I don't know if this breaks things downstream

I am not too concerned: this only adds new functionality in a way that should not affect existing functionality.

> I noticed Annoy has the dot product distance 

I will admit to having missed that.

**Comment by benjamin-james** [comment]
> This looks very good!
> 
> Would you mind adding a ChangeLog entry (date/name/email with two spaces between them, tab to '*' describing file changed and how, one can group as needed).

Perfect, just pushed commit.

**Comment by eddelbuettel** [comment]
Turns out we forgot one microaspect of an added alias for the Rd file.  I just took care of that in 6446e7b469c7c77d3eaa39037fe33d63332107f1.

**Pull request merged**